[NFC][ntuple] add v6.34 release notes (#16887)
root-project/root: README/ReleaseNotes/v634/index.md

## I/O Libraries

## RNTuple Libraries

> * The first version of the on-disk binary format is finalized. Future versions of ROOT will be able to read back

Reviewer: ```suggestion
* The first version of the `RNTuple` on-disk binary format is finalized. Future versions of ROOT will be able to read back
```
let's be pedantic to allow easy accurate copy/paste :)

  `RField<float|double>::SetTruncated()` (truncated mantissa) and `RField<float|double>::SetQuantized()`
  (scaled integer representation).
* Link RNTuple self-description to the common ROOT streamer infrastructure. As a result, `TFile::MakeProject()`
  properly creates header files for classes used in RNTuple data.
* First version of the new `RNTupleProcessor` class. The `RNTupleProcessor` will support iteration of composed RNTuple data sets (comparable to and improving upon TTree friends and chains). This release supports chained (vertically composed) RNTuples. Other types of concatenations will be added in subsequent releases.
* Support for cluster staging in the `RNTupleParallelWriter`. Cluster staging enables users to enforce a certain
  logical cluster ordering in the presence of parallel cluster writing.
* Support for Direct I/O for writing. This gives access to the peak performance of modern NVMe drives.
* Support for a "streamer field" that can wrap classic ROOT I/O serialized data for RNTuple in cases where native
  RNTuple support is not possible (e.g., recursive data structures). Use of the streamer field can be enforced

>   through the LinkDef option `rntupleStreamerMode(true)`.

Reviewer: Consider adding
```suggestion
  through the LinkDef option `rntupleStreamerMode(true)`.  This features is similar to the unsplit/level-0 split branch in `TTree`
```

## I/O Libraries

## RNTuple Libraries

* The first version of the on-disk binary format is finalized. Future versions of ROOT will be able to read back
  RNTuple data written as of this release. Please note that this version breaks compatibility with experimental RNTuple
  data written with releases up to v6.34. Please also note that the RNTuple API is not yet moving out of
  `ROOT::Experimental`.
* Support for low-precision on-disk floating point representation. This can be enabled through
  `RField<float|double>::SetTruncated()` (truncated mantissa) and `RField<float|double>::SetQuantized()`
  (scaled integer representation).
* Link RNTuple self-description to the common ROOT streamer infrastructure. As a result, `TFile::MakeProject()`
  properly creates header files for classes used in RNTuple data.
* First version of the new `RNTupleProcessor` class. The `RNTupleProcessor` supports iteration of joined RNTuple

>   data sets (à la friends and chains).

Reviewer 2: This release only supports chains for now, I think we should clarify that.

Reviewer 2: My suggested change: 
```suggestion
* First version of the new `RNTupleProcessor` class. The `RNTupleProcessor` will support iteration of composed RNTuple data sets (comparable to and improving upon TTree friends and chains). This release supports chained (vertically composed) RNTuples. Other types of concatenations will be added in subsequent releases.
```
